extensions/sql-migration/src/wizard/databaseBackupPage.ts

```diff
@@ -525,89 +525,105 @@ export class DatabaseBackupPage extends MigrationWizardPage {
-			this._networkShareContainerSubscriptionDropdown.values = await this.migrationStateModel.getSubscriptionsDropdownValues();
-			this._blobContainerSubscriptionDropdown.values = await this.migrationStateModel.getSubscriptionsDropdownValues();
-			this.migrationStateModel.databaseBackup.subscription = this.migrationStateModel.getSubscription(0);
-		} catch (error) {
-			this.migrationStateModel._storageAccounts = undefined!;
-		} finally {
-			await this.loadNetworkShareStorageDropdown();
-			await this.loadFileShareStorageDropdown();
-			await this.loadblobStorageDropdown();
-			this._networkShareContainerSubscriptionDropdown.loading = false;
-			this._fileShareSubscriptionDropdown.loading = false;
-			this._blobContainerSubscriptionDropdown.loading = false;
+		if (!this.migrationStateModel.databaseBackup.subscription) {
+			this._networkShareContainerSubscriptionDropdown.loading = true;
+			this._fileShareSubscriptionDropdown.loading = true;
+			this._blobContainerSubscriptionDropdown.loading = true;
+			try {
+				this._fileShareSubscriptionDropdown.values = await this.migrationStateModel.getSubscriptionsDropdownValues();
+				this._networkShareContainerSubscriptionDropdown.values = await this.migrationStateModel.getSubscriptionsDropdownValues();
+				this._blobContainerSubscriptionDropdown.values = await this.migrationStateModel.getSubscriptionsDropdownValues();
+				this.migrationStateModel.databaseBackup.subscription = this.migrationStateModel.getSubscription(0);
+			} catch (error) {
+				console.log(error);
+			} finally {
+				this.migrationStateModel._storageAccounts = undefined!;
+				await this.loadNetworkShareStorageDropdown();
+				await this.loadFileShareStorageDropdown();
+				await this.loadblobStorageDropdown();
+				this._networkShareContainerSubscriptionDropdown.loading = false;
+				this._fileShareSubscriptionDropdown.loading = false;
+				this._blobContainerSubscriptionDropdown.loading = false;
+			}
 		}
 	}
 
 	private async loadNetworkShareStorageDropdown(): Promise<void> {
-		this._networkShareContainerStorageAccountDropdown.loading = true;
-		try {
-			this._networkShareContainerStorageAccountDropdown.values = await this.migrationStateModel.getStorageAccountValues(this.migrationStateModel.databaseBackup.subscription);
-			this.migrationStateModel.databaseBackup.storageAccount = this.migrationStateModel.getStorageAccount(0);
-		} finally {
-			this._networkShareContainerStorageAccountDropdown.loading = false;
+		if (!this.migrationStateModel.databaseBackup.storageAccount) {
+			this._networkShareContainerStorageAccountDropdown.loading = true;
+			try {
+				this._networkShareContainerStorageAccountDropdown.values = await this.migrationStateModel.getStorageAccountValues(this.migrationStateModel.databaseBackup.subscription);
+				this.migrationStateModel.databaseBackup.storageAccount = this.migrationStateModel.getStorageAccount(0);
```

Comment:
Is setting the storage account here necessary? I would think you'd be hooking into the onChanged event for these to update the model value instead

(same for the other similar logic in the other functions)